proposals/1113_ldap-ad-web-ui/README.md

```diff
@@ -0,0 +1,207 @@
+Users will still have the option to override this priority with the `--upstream-identity-provider-flow` flag,
+but that flag will not be required.
+
+#### Tests
+
+Chromedriver browser based integration tests will be needed to ensure that a user can log in from a web-based app
+by entering their ldap credentials into the web page, as well as unit tests.
+
+With the pinniped cli:
+- succeeds with correct username and password
+- fails with incorrect username, shows useful but nonspecific error message
+- fails with incorrect password, shows useful but nonspecific error message
+- with tty access, prompts for username and password on the cli
+- without tty access, opens a browser
+- without tty access, if the form post fails, don't ask user to copy and paste the authcode (we already know you have no tty to paste it into...)
+Once dynamic clients are implemented:
+- fails when attempting to pass username/password as headers on requests to the authorize endpoint
+- tests of the rest of the dynamic client functionality that should be detailed as part of that proposal
+
+#### New Dependencies
+This should be kept to a very simple HTML page with minimal, clean CSS styling.
+Javascript should be avoided.
+The styling should match the [form post html page](https://github.com/vmware-tanzu/pinniped/tree/main/internal/oidc/provider/formposthtml)
+as much as possible, we should reuse some of the existing css and add to it to keep the style consistent.
+
+#### Observability Considerations
+* Logging login attempts at higher log levels.
+
+#### Security Considerations
+* Preventing LDAP injection attacks: this should be done server-side using our existing
+  string escaping.
+* CSRF protection via a CSRF cookie: this should be similar to the way it is done for the
+  OIDCIdentityProvider today
+* The new UI page must be HTTPS.
+
+#### Documentation Considerations
+This new feature will require documentation to explain how to configure it and to publicise that it is available.
+This should include:
+* A blog post describing the feature
+* Website documentation in the form of a how-to guide
+
+### Other Approaches Considered
+Today, users can configure Dex if they want a web-based LDAP login.
+This introduces complexity because they have to install, configure and
+maintain both Pinniped and Dex in order to use this feature. It also means
+that users do not benefit from the opinionated `ActiveDirectoryIdentityProvider`
+config because Dex does not have an equivalent.
+
+## Open Questions
+* What is the format for the URL? (`issuer/some/path`? Something else?) Can we make it so we can reuse the existing cert,
+  or will we need a new wildcard cert?
+* Currently we have little validation on branding requirements. Is specifying the idp name enough for users to understand
+  how to log in?
+* How many users will be blocked on using this feature until they can have a company name and logo on the login page?
```

Comment:
Can we move these to the "Answered Questions" section and write down our current answers?